### PR TITLE
KVS: Add unit tests for media sources

### PR DESCRIPTION
**Pull request by sirknightj**
*Issue #, if available:*

*Description of changes:*
Added unit tests for the newly-added [ImageFileMediaSource](https://github.com/aws-amplify/aws-sdk-android/pull/2596) and the existing AndroidCameraMediaSource. Tests mainly check for whether the media source configuration fields are set correctly.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-kinesisvideo/src/test/java/com/amazonaws/mobileconnectors/kinesisvideo/mediasource/file/ImageFileMediaSourceConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        assertThat(config.getFilenameFormat()).isEqualTo(FILE_NAME_FORMAT);
+        assertThat(config.getStartFileIndex()).isEqualTo(START_FILE_INDEX);
+        assertThat(config.getEndFileIndex()).isEqualTo(END_FILE_INDEX);
+    }
+
+    @Test
+    public void testEmptyFilePathIsValid() {
+        ImageFileMediaSourceConfiguration config = new ImageFileMediaSourceConfiguration.Builder()
+                .fps(FPS)
+                .dir("")
```
Comment (unicornss) [inline aws-android-sdk-kinesisvideo/src/test/java/com/amazonaws/mobileconnectors/kinesisvideo/mediasource/file/ImageFileMediaSourceConfigurationTest.java:54]: are we considering to read the images from current directory in this setting?

Comment (sirknightj) [reply aws-android-sdk-kinesisvideo/src/test/java/com/amazonaws/mobileconnectors/kinesisvideo/mediasource/file/ImageFileMediaSourceConfigurationTest.java:54]: Yes. See https://github.com/awslabs/aws-sdk-android-samples/blob/dab285be113a68daa2a72f03de52c1b4b679a2a6/AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/fragment/FramesFragment.java#L127 for more info

**Review by rjuliano — APPROVED** [review]
LGTM

**Review by raphkim — CHANGES_REQUESTED** [review]
Added unit tests appear to be failing in circleCI. Care to take a look before we merge this?
Screen Shot 2021-09-08 at 12 18 00 PM

**Comment by unicornss** [comment]
Seems like Java version https://41264-21634100-gh.circle-artifacts.com/0/%7E/test_results/aws-android-sdk-kinesisvideo/testDebugUnitTest/classes/com.amazonaws.mobileconnectors.kinesisvideo.mediasource.android.AndroidCameraMediaSourceTest.html 

java.lang.UnsupportedOperationException: Failed to create a Robolectric sandbox: Android SDK 29 requires Java 9 (have Java 8)

**Comment by raphkim** [comment]
You will have to specify the appropriate SDK in robolectric properties file in the test resources directory when adding robolectric dependency in this project. For reference: https://github.com/aws-amplify/aws-sdk-android/tree/main/aws-android-sdk-kinesis/src/test/resources.

**Review by raphkim — APPROVED** [review]

**Pull request merged**